src/sapling/sapling_operation.cpp

```diff
@@ -208,6 +227,25 @@ void SaplingOperation::setFromAddress(const libzcash::SaplingPaymentAddress& _pa
 
 OperationResult SaplingOperation::loadUtxos(TxValues& txValues)
 {
+    // If the user has selected coins to spend then, directly load them.
+    // The spendability, depth and other checks should have been done on the user selection side,
+    // no need to do them again.
+    if (coinControl && coinControl->HasSelected()) {
+        std::vector<OutPointWrapper> vCoins;
+        coinControl->ListSelected(vCoins);
```

Comment:
Pretty nice. Guess this is the first step to get to the removal of `coinControl` from the arguments passed to AvailableCoins, that we discussed back in https://github.com/PIVX-Project/PIVX/pull/1859#issuecomment-696235724

Reply:
I actually totally forgot about that nice comment hehe. We need a bigger whiteboard..
This was just an idea that has while was coding the entire flow.

We can definitely remove `coinControl` from `AvailableCoin` in a subsequent PR.